Friendly-plans/app/src/androidTest/java/pg/autyzm/friendly_plans/TaskCreateActivityTest.java

```diff
@@ -87,22 +108,114 @@ public void When_AddingNewTask_Expect_NewTaskAddedToDB() throws InterruptedExcep
+
+        assertThat(taskTemplates.size(), is(1));
+        assertThat(taskTemplates.get(0).getName(), is(EXPECTED_NAME));
+        assertThat(taskTemplates.get(0).getDurationTime(), is(EXPECTED_DURATION));
+    }
+
+    @Test
+    public void When_SettingPicture_Expect_PictureNameIsDisplayed()
+            throws InterruptedException, IOException {
+        setTestPicture();
+        List<Asset> assets = assetRepository.getAll();
+
+        onView(withId(R.id.id_et_task_picture))
+                .check(matches(withText(assets.get(0).getFilename())));
+    }
+
+    @Test
+    public void When_AddingNewTaskWithPicture_Expect_NewTaskAddedToDB()
+            throws InterruptedException, IOException {
+        onView(withId(R.id.id_et_task_name))
+                .perform(replaceText(EXPECTED_NAME));
+        closeKeyboard();
+
+        onView(withId(R.id.id_et_task_duration_time))
+                .perform(replaceText(EXPECTED_DURATION_TXT));
         closeKeyboard();
 
+        setTestPicture();
+
         onView(withId(R.id.id_btn_task_next))
                 .perform(click());
 
+        List<Asset> assets = assetRepository.getAll();
         List<TaskTemplate> taskTemplates = taskTemplateRepository.get(EXPECTED_NAME);
         idToDelete = taskTemplates.get(0).getId();
 
+        assertThat(assets.size(), is(1));
         assertThat(taskTemplates.size(), is(1));
         assertThat(taskTemplates.get(0).getName(), is(EXPECTED_NAME));
-        assertThat(taskTemplates.get(0).getDurationTime(), is(1));
+        assertThat(taskTemplates.get(0).getDurationTime(), is(EXPECTED_DURATION));
+        assertThat(taskTemplates.get(0).getPictureId(), is(assets.get(0).getId()));
     }
 
     private void closeKeyboard() throws InterruptedException {
         closeSoftKeyboard();
         Thread.sleep(1000);
     }
+
+    private void setTestPicture() throws IOException, InterruptedException {
+        File testPicture = createTestPicture();
+        chooseTestPicture(testPicture);
+        addSafeCopyToTestFiles();
+    }
+
+    private File createTestPicture() throws IOException {
+        File testPicture = new File(internalStorage, TEST_PICTURE_NAME);
+        FileUtils.writeStringToFile(testPicture, "Test");
+        testFiles.add(testPicture);
+        return testPicture;
+    }
+
+    private void chooseTestPicture(File testPicture) throws InterruptedException {
+        final Intent data = new Intent();
+        data.putExtra(FilePickerActivity.RESULT_FILE_PATH, testPicture.getAbsolutePath());
+        final TaskContainerFragment fragment = getFragment();
+        activityRule.getActivity().runOnUiThread(new Runnable() {
+            public void run() {
+                fragment.onActivityResult(FilePickerProxy.PICK_FILE_REQUEST,
+                        FilePickerActivity.RESULT_OK, data);
+            }
+        });
+        Thread.sleep(1000);
+    }
+
+    private void addSafeCopyToTestFiles() {
```

Comment:
It's great idea! After applying this changes test code looks much cleaner and assets' managing part is reusable :)